packages/react-core/src/components/Checkbox/__tests__/Checkbox.test.tsx

```diff
@@ -6,91 +6,239 @@ import userEvent from '@testing-library/user-event';
-    const myMock = jest.fn();
-    global.console = { ...global.console, error: myMock };
+  expect(screen.getByRole('checkbox')).toBeDisabled();
+});
 
-    render(<Checkbox id={undefined} />);
-    expect(myMock).toHaveBeenCalled();
-  });
+test('Does not set the checkbox as required by default', () => {
+  render(<Checkbox id="test-id" />);
 
-  test('renders component wrapper as span', () => {
-    const { container } = render(
-      <Checkbox component="span" label="label" aria-labelledby="labelId" id="check" isChecked aria-label="check" />
-    );
-    const span = container.querySelector('span');
-    expect(span).toHaveClass(styles.check);
-  });
+  expect(screen.getByRole('checkbox')).not.toBeRequired();
+});
+
+test(`Sets the checkbox as required when isRequired is passed`, () => {
+  render(<Checkbox id="test-id" isRequired />);
+
+  expect(screen.getByRole('checkbox')).toBeRequired();
+});
+
+test('Does not set the checkbox as checked by default', () => {
+  render(<Checkbox id="test-id" />);
+
+  expect(screen.getByRole('checkbox')).not.toBeChecked();
+});
+
+test(`Sets the checkbox as checked when isChecked is passed`, () => {
+  render(<Checkbox id="test-id" isChecked />);
+
+  expect(screen.getByRole('checkbox')).toBeChecked();
+});
+
+test(`Sets the checkbox as checked when checked is passed`, () => {
+  render(<Checkbox id="test-id" checked />);
+
+  expect(screen.getByRole('checkbox')).toBeChecked();
+});
+
+test(`Calls onChange when the checkbox is clicked`, async () => {
+  const user = userEvent.setup();
+  const onChange = jest.fn();
+  render(<Checkbox id="test-id" onChange={onChange} />);
+
+  await user.click(screen.getByRole('checkbox'));
+
+  expect(onChange).toHaveBeenCalledTimes(1);
+});
+
+test('Does not call onChange when the checkbox is not clicked', () => {
+  const onChange = jest.fn();
+  render(<Checkbox id="test-id" onChange={onChange} />);
+
+  expect(onChange).not.toHaveBeenCalled();
+});
+
+test(`Calls onChange with the event and the checked value when the checkbox is clicked`, async () => {
+  const user = userEvent.setup();
+  const onChange = jest.fn();
+  render(<Checkbox id="test-id" onChange={onChange} />);
+
+  await user.click(screen.getByRole('checkbox'));
+
+  expect(onChange).toHaveBeenCalledWith(expect.objectContaining({ type: 'change' }), true);
+});
+
+test(`Calls onChange with the event and the checked value when the checkbox is clicked and isChecked is passed`, async () => {
+  const user = userEvent.setup();
+  const onChange = jest.fn();
+  render(<Checkbox id="test-id" isChecked onChange={onChange} />);
+
+  await user.click(screen.getByRole('checkbox'));
+
+  expect(onChange).toHaveBeenCalledWith(expect.objectContaining({ type: 'change' }), false);
+});
+
+test('Does not render a label by default', () => {
+  render(<Checkbox id="test-id" />);
+
+  expect(screen.queryByLabelText('test label')).not.toBeInTheDocument();
+});
```

Comment:
This would apply to the description and body tests as well, but I'm always unsure of what tests like this are `expect`ing just because technically they're passing, but they'd also pass even if `label="another test label"` were passed in. So less that a label doesn't render and more that a label with this specific test doesn't render.

At the same time I'm not sure how pretty an alternative would be. Using something like `getByRole(checkbox).parentElement.querySelector(.${styles.checkLabel})` may work but is lengthy.

Probably splitting hairs too much so I don't think I'd really block over it, but curious what you think.

Reply:
Yeah I don't have a hardline stance on this and happy to remove if you think they're unnecessary, but my reasoning for doing these kinds of tests is that it ensure that assertion in the following test (where we expect the label to exist) is actually valid and not giving a false positive.

Reply:
Since it'd accept a regex, what about `expect(screen.queryByLabelText(/\w+/)).not.toBeInTheDocument();`? We don't really care what the label text actually is, just whether it's rendered, so this would result in a failure if any sort of label text were to be found (rather than currently a failure only occurs if label text that is "test label" is found)

Reply:
Yeah I'm not against that idea at all.